src/main/java/dao/CurrencyDao.java

```diff
@@ -50,37 +52,36 @@ public List<Currency> getSpecificCurrency(String code) throws SQLException {
         return specificCurrency;
     }
 
-    public List<Currency> postNewCurrency(String code, String name, String sign) throws SQLException {
-        List<Currency> newCurrency = new ArrayList<>();
-        final String queryToAddNewCurrency = "INSERT INTO Currencies (code, name, sign) VALUES (?, ?, ?)";
+    public Currency addNewCurrency(String code, String fullName, String sign) throws SQLException {
 
+        final String queryToAddNewCurrency = "INSERT INTO Currencies (Code, fullName, Sign) VALUES (?, ?, ?)";
+
+        Currency currency = null;
         try (Connection connection = DataConfig.getDataSource().getConnection()) {
             PreparedStatement prepareStatement = connection.prepareStatement(queryToAddNewCurrency, Statement.RETURN_GENERATED_KEYS);
 
             prepareStatement.setString(1, code);
-            prepareStatement.setString(2, name);
+            prepareStatement.setString(2, fullName);
             prepareStatement.setString(3, sign);
 
             int rowsAffected = prepareStatement.executeUpdate();
 
             if (rowsAffected > 0) {
                 try (ResultSet rs = prepareStatement.getGeneratedKeys()) {
                     if (rs.next()) {
-                        Currency currency = Currency.builder()
+                        currency = Currency.builder()
                                 .id(rs.getLong(1))
                                 .code(code)
-                                .fullName(name)
+                                .fullName(fullName)
                                 .sign(sign)
                                 .build();
-                        newCurrency.add(currency);
                     }
                 }
             }
         } catch (SQLException e) {
             e.printStackTrace();
```

Comment:
```suggestion
            throw new RuntimeException(e)
```
Попробуй поиграть с этим и посмотри какой у тебя стек трейс будет в случае как есть сейчас. Какой будет в том случае, который я предложил и в случае throw new RuntimeException() не передавая туда e

И так везде. Отпиши потом мне, в чем ты понял разницу